### ISSUE TITLE
Block third-party cookies in Chrome breaks scroll option and localStorage access

### ISSUE THREAD

**Issue by willhains** [simov/markdown-viewer#70]
Turning on the "scroll" option in the "Content" tab breaks syntax highlighting.

**Comment by simov** [comment]
You have to disable the [Content Security Policy](https://github.com/simov/markdown-viewer#disable-content-security-policy) for that origin in the Advanced Options page.

The problem is that the scroll logic is executed before the syntax highlighting and it fails because it is blocked by the CSP header present on that origin.

**Comment by willhains** [comment]
I’m seeing this problem with local files. There doesn’t appear to be a “Disable Content Security Policy” option on the `file://` origin.

**Comment by simov** [comment]
That's very weird. The scroll option is there from the beginning and there was never a problem with it. Probably you could open up the dev tools and see if there is an error message in the console.

**Comment by willhains** [comment]
Tried that, and as you predicted, I see an error in the console:
```
Uncaught DOMException: Failed to read the ‘localStorage’ property from ‘Window’: Access is denied for this document.
  at init (file:///......../content/content.js:165:38)
  at race (file:///......../content/content.js:182:7)
  at race (file:///........../content/content.js:150:7)
  etc.
```

**Comment by simov** [comment]
Yes, `localStorage` is used to store the scroll position between page reloads. Do you see rendered markdown at all, or only the syntax highlighting is missing?

Also I have a file in `/content/content.js` in the extension responsible for that logic, but the line numbers in your error message are slightly off. Not that I have ever experienced such issue, but are you sure you are using the latest version?

**Comment by willhains** [comment]
> Do you see rendered markdown at all, or only the syntax highlighting is missing?

Everything renders correctly except code block syntax highlighting, as far as I can tell.

> are you sure you are using the latest version?

It says I am using version 3.6, which seems to match the version in the Chrome Web Store.

When I run with the Developer Console open, execution pauses at line 165, column 38 (which is `localStorage.getItem(key)`).

**Comment by simov** [comment]
Right, my bad, that's the correct line, I just have some changes to that file locally. So that's very interesting indeed, for some reason access to localStorage is not allowed on file:/// URLs, but i have no idea why.

Which browser version do you use and which OS?

**Comment by willhains** [comment]
I am using markdown-viewer on Chrome v27.0.3578.98 (64-bit) on Windows 7 (64-bit).

**Comment by simov** [comment]
Just tried with Chrome 71 on Windows 7, everything works fine. In fact, the syntax highlighting and the scroll option are there from the initial release of the extension a few years ago, and you are the first one reporting such issue.

Can it be some custom flag set in chrome://flags? Also try disabling all other extensions that you have.

**Comment by simov** [comment]
I'm running out of ideas. That might remain a mystery.

**Comment by willhains** [comment]
I found the problem.

I googled for the error message above, and the [first hit](
https://www.chromium.org/for-testers/bug-reporting-guidelines/uncaught-securityerror-failed-to-read-the-localstorage-property-from-window-access-is-denied-for-this-document) says the exception is thrown when the “Block third-party cookies” option is set in Content Settings.

Sure enough, turning this option off fixes the problem.

I’m glad it’s now working, but I’d really prefer to have that option turned on! Do you know which third-party site’s cookies are causing the problem? I am probably fine to whitelist just that one site, and have the general blocking feature turned on.

**Comment by simov** [comment]
Well, it will take some time to get to it, but I'll let you know when I have something ready.

**Comment by simov** [comment]
Published a fix in v3.7, you should receive the update shortly.

**Comment by willhains** [comment]
Confirmed this works now. Thanks!! 😊
